eth2near/eth2near-block-relay/eth-proof-extractor.js

```diff
@@ -67,6 +68,65 @@ class EthProofExtractor {
   }
 }
 
+async function ethToNearFindProof ({ lockedEventRaw, ethNodeUrl }) {
+  const robustWeb3 = new RobustWeb3(ethNodeUrl)
+  const web3 = robustWeb3.web3
+  try {
+    const lockedEvent = JSON.parse(lockedEventRaw) || {}
+    const extractor = new EthProofExtractor()
+    extractor.initialize(ethNodeUrl)
+
+    const receipt = await extractor.extractReceipt(lockedEvent.transactionHash)
+    const block = await extractor.extractBlock(receipt.blockNumber)
+    const tree = await extractor.buildTrie(block)
+    const proof = await extractor.extractProof(
+      web3,
+      block,
+      tree,
+      receipt.transactionIndex
+    )
+    extractor.destroy()
+
+    let txLogIndex = -1
+
+    let logFound = false
+    let log
+    for (const receiptLog of receipt.logs) {
+      txLogIndex++
+      const blockLogIndex = receiptLog.logIndex
+      if (blockLogIndex === lockedEvent.logIndex) {
+        logFound = true
+        log = receiptLog
+        break
+      }
+    }
+    if (logFound) {
+      const logEntryData = logFromWeb3(log).serialize()
+      const receiptIndex = proof.txIndex
+      const receiptData = receiptFromWeb3(receipt).serialize()
+      const headerData = proof.header_rlp
+      const _proof = []
```

Comment:
Can it have a better name?

Reply:
This is taken from `from-near.js`. We can rename it.

Reply:
Done.